# Indian Start-up Investment Analysis (2018 - 2021)

### Aim:
To assess the investment potential and attractiveness of the Indian startup ecosystem and provide recommendations for optimal course of action

### Objectives:
 
1. To assess the overall attractiveness of the Indian startup ecosystem based on funding trends and investor activity from 2018 to 2021.
2. To identify key sectors with high potential for investment based on their funding attractiveness and growth prospects.
3. To evaluate the investment opportunities across different stages of startup development and their risk-return profiles.
4. To analyze the geographical distribution of startups and funding to identify strategic investment locations and regional investment disparities.
5. To determine the correlation between funding amounts received by startups and their subsequent performance, providing insights into potential returns on investment and success rates.

### Business Questions:
1. How have the trends in funding amounts varied among different sectors of Indian startups from 2018 to 2021, and what implications do these trends have for investors seeking to optimize their investment strategies in the Indian startup ecosystem?

2. What were the top sectors in the Indian startup ecosystem that received the highest total funding from 2018 to 2021, and what factors contributed to their success in attracting investment compared to other sectors?

    This question delves into not just the identification of the sectors but also seeks to understand the underlying reasons for their success in attracting funding, which could include factors like market demand, technological innovation, regulatory environment, and investor interest.

3. What are the specific funding trends across different stages of startup development (early-stage, growth-stage, late-stage) within the Indian startup ecosystem, and how do factors such as sector, geographic location, and investor type influence these trends?
This refined question allows for a more comprehensive analysis of funding trends by considering additional dimensions such as:
Startup Stages: Early-stage, growth-stage, and late-stage.
Sector: Identifying which sectors are receiving funding at various stages.
Geographic Location: Understanding regional variations in funding.
Investor Type: Examining the role of different types of investors (e.g., venture capital, angel investors, private equity) in funding trends.

4. What specific geographic factors (such as infrastructure, talent pool, economic policies, and market access) influence the correlation between the geographical location of startups and the funding they received within the Indian startup ecosystem, and how do these factors vary across different regions?

5. What specific factors (such as the amount and stage of funding, investor involvement, business model, and market conditions) influence the relationship between funding amounts and the subsequent success or failure of startups within the Indian ecosystem, and how do these factors vary across different sectors and stages of startup development?

### Hypothesis to Test:
 
Given the goal of assessing the investment potential in the Indian startup ecosystem, we hypothesize that:
 
**Null Hypothesis (H0)**: There is no clear pattern in the funding received by Indian startups from 2018 to 2021, and factors like sector, stage, location, and funding amount do not affect startup success.

**Alternative Hypothesis (H1)**: There is a clear pattern in the funding received by Indian startups from 2018 to 2021, and factors like sector, stage, location, and funding amount affect startup success.

In [1]:
# import relevant packages
import pyodbc
from dotenv import dotenv_values
import pandas as pd
import warnings
import numpy as np

warnings.filterwarnings('ignore')


#### Connect to server for 2020 and 2021 datasets

In [2]:
# load environment variables from .env file into a dictionary
environment_variables = dotenv_values('.env')

# Get the values for the credentials from .env file
database=environment_variables.get("DATABASE")
server=environment_variables.get("SERVER")
login=environment_variables.get("LOGIN")
password=environment_variables.get("PASSWORD")

# create a connection string
connection_string=f"DRIVER={{SQL Server}};SERVER={server};DATABASE={database};UID={login};PWD={password}"

In [3]:
# create connection using the pyodbc method 

connection = pyodbc.connect(connection_string)

#### Select tables of interest from the Database

In [4]:
# selecting tables from Database
db_query = ''' SELECT *
            FROM INFORMATION_SCHEMA.TABLES
            WHERE TABLE_TYPE = 'BASE TABLE' '''

#### View tables of interest from the Database for verification purposes

In [5]:
# call selected table from SQL Database
ata=pd.read_sql(db_query, connection)

ata

TABLE_CATALOG TABLE_SCHEMA               TABLE_NAME  TABLE_TYPE
0         dapDB          dbo  LP1_startup_funding2021  BASE TABLE
1         dapDB          dbo  LP1_startup_funding2020  BASE TABLE

In [6]:
# Call DataFrame to understand DataFrame details for 2020
query= "SELECT * FROM dbo.LP1_startup_funding2020"
data_2020 =pd.read_sql(query, connection)

data_2020.head()

Company_Brand  Founded HeadQuarter              Sector  \
0    Aqgromalin   2019.0     Chennai            AgriTech   
1      Krayonnz   2019.0   Bangalore              EdTech   
2  PadCare Labs   2018.0        Pune  Hygiene management   
3         NCOME   2020.0   New Delhi              Escrow   
4    Gramophone   2016.0      Indore            AgriTech   

                                        What_it_does  \
0                       Cultivating Ideas for Profit   
1  An academy-guardian-scholar centric ecosystem ...   
2   Converting bio-hazardous waste to harmless waste   
3                       Escrow-as-a-service platform   
4  Gramophone is an AgTech platform enabling acce...   

                                            Founders  \
0                    Prasanna Manogaran, Bharani C L   
1                   Saurabh Dixit, Gurudutt Upadhyay   
2                                    Ajinkya Dhariya   
3                                      Ritesh Tiwari   
4  Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   

                              Investor    Amount     Stage column10  
0                      Angel investors  200000.0      None     None  
1                      GSF Accelerator  100000.0  Pre-seed     None  
2                       Venture Center       NaN  Pre-seed     None  
3  Venture Catalysts, PointOne Capital  400000.0      None     None  
4  Siana Capital Management, Info Edge  340000.0      None     None

In [7]:
#check the shape
data_2020.shape    

(1055, 10)

In [8]:
# Check the description of the DataFrame

data_2020.describe().T

count          mean           std      min        25%        50%  \
Founded  842.0  2.015363e+03  4.097909e+00   1973.0     2014.0     2016.0   
Amount   801.0  1.130430e+08  2.476635e+09  12700.0  1000000.0  3000000.0   

                75%           max  
Founded      2018.0  2.020000e+03  
Amount   11000000.0  7.000000e+10

#### Observation
On average founding year is around 2015, with a standard deviation of about 4.1 years indicating that the founding years are fairly concentrated around the mean (2015). This suggest that most companies were founded within a narrow time frame around 2015 in the dataset_2020.

In [9]:
# check data information to understand data structure
data_2020.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1055 entries, 0 to 1054
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company_Brand  1055 non-null   object 
 1   Founded        842 non-null    float64
 2   HeadQuarter    961 non-null    object 
 3   Sector         1042 non-null   object 
 4   What_it_does   1055 non-null   object 
 5   Founders       1043 non-null   object 
 6   Investor       1017 non-null   object 
 7   Amount         801 non-null    float64
 8   Stage          591 non-null    object 
 9   column10       2 non-null      object 
dtypes: float64(2), object(8)
memory usage: 82.6+ KB


In [10]:
# Calculate the number of missing values
missing_values_count = data_2020.isna().sum()

# Calculate the percentage of missing values
missing_values_percentage = (missing_values_count / len(data_2020)) * 100

# Combine the two into a DataFrame for better readability
missing_values_summary = pd.DataFrame({
    'Missing Values': missing_values_count,
    'Percentage': missing_values_percentage
})

# Print the summary
print(missing_values_summary)


               Missing Values  Percentage
Company_Brand               0    0.000000
Founded                   213   20.189573
HeadQuarter                94    8.909953
Sector                     13    1.232227
What_it_does                0    0.000000
Founders                   12    1.137441
Investor                   38    3.601896
Amount                    254   24.075829
Stage                     464   43.981043
column10                 1053   99.810427


#### Statistical Observations on missing values
The dataset_2020 is highly complete for **Company_Brand** and **What_it_does**, with no missing values, ensuring consistent descriptive information for each company. Conversely, the **Stage** column faces the most significant issue, with 43.98% missing values, which will hamper the ability to analyze the development stages of the companies comprehensively. However, there are only minimal missing data in the **Sector**, **Founders**, and **Investor** columns (1.23%, 1.14%, and 3.60% respectively).

In [49]:
#Import relevant package
from sklearn.impute import SimpleImputer

# Calculate the count and percentage of missing values for each column
missing_values_count = data_2020.isna().sum()
missing_values_percentage = (missing_values_count / len(data_2020)) * 100

# Combine the count and percentage into a DataFrame for better readability
missing_values_summary = pd.DataFrame({
    'Missing Values': missing_values_count,
    'Percentage': missing_values_percentage
})

# Print the missing values summary
print("Missing Values Summary:\n", missing_values_summary)

# Define a function to replace string 'None' with 'Undisclosed' in specified columns
def replace_none_with_undisclosed(df, columns):
    for column in columns:
        # Replace 'None' with 'Undisclosed'
        df[column] = df[column].astype(str).replace('None', 'Undisclosed')
    return df

# Apply the function to the relevant columns to standardize missing data representation
columns_to_replace = ['HeadQuarter', 'Stage', 'Sector', 'Founders', 'Investor']
data_2020 = replace_none_with_undisclosed(data_2020, columns_to_replace)

# Create a SimpleImputer object to fill 'Undisclosed' values with the most frequent value
categorical_imputer = SimpleImputer(strategy='most_frequent')

# Apply the imputer to fill 'Undisclosed' values in the specified categorical columns
data_2020[columns_to_replace] = categorical_imputer.fit_transform(data_2020[columns_to_replace])

# Create a SimpleImputer object for numerical columns 'Founded' and 'Amount' using the mean strategy
numerical_columns = ['Founded', 'Amount']
numerical_imputer = SimpleImputer(strategy='mean')

# Apply the imputer to fill missing values in the numerical columns
data_2020[numerical_columns] = numerical_imputer.fit_transform(data_2020[numerical_columns])


# Verify the changes by printing the updated missing values summary
missing_values_count_updated = data_2020.isna().sum()
missing_values_summary_updated = pd.DataFrame({
    'Missing Values': missing_values_count_updated,
    'Percentage': (missing_values_count_updated / len(data_2020)) * 100
})




Missing Values Summary:
               Missing Values  Percentage
Company                    0    0.000000
Founded                    0    0.000000
HeadQuarter                0    0.000000
Sector                     0    0.000000
What_it_does               0    0.000000
Founders                   0    0.000000
Investor                   0    0.000000
Amount                     0    0.000000
Stage                      0    0.000000
column10                1053   99.810427
year_founded               0    0.000000


In [12]:
# Check for duplicate rows
duplicates = data_2020.duplicated()

# Count duplicate rows
num_duplicates = duplicates.sum()
print(f"Number of duplicate rows: {num_duplicates}")

# Print duplicate rows
duplicate_rows = data_2020[duplicates]
print("Duplicate rows:")
print(duplicate_rows)

# Drop duplicate rows
data_2020_cleaned = data_2020.drop_duplicates()

# Display the number of rows before and after removing duplicates
print(f"Number of rows before removing duplicates: {data_2020.shape[0]}")
print(f"Number of rows after removing duplicates: {data_2020_cleaned.shape[0]}")


Number of duplicate rows: 3
Duplicate rows:
    Company_Brand  Founded HeadQuarter                 Sector  \
145     Krimanshi   2015.0     Jodhpur  Biotechnology company   
205         Nykaa   2012.0      Mumbai              Cosmetics   
362        Byju’s   2011.0   Bangalore                 EdTech   

                                          What_it_does         Founders  \
145  Krimanshi aims to increase rural income by imp...     Nikhil Bohra   
205  Nykaa is an online marketplace for different b...    Falguni Nayar   
362  An Indian educational technology and online tu...  Byju Raveendran   

                                           Investor        Amount  \
145  Rajasthan Venture Capital Fund, AIM Smart City  6.000000e+05   
205                        Alia Bhatt, Katrina Kaif  1.130430e+08   
362           Owl Ventures, Tiger Global Management  5.000000e+08   

           Stage column10  
145         Seed     None  
205  Undisclosed     None  
362  Undisclosed     None  
Numbe

In [59]:
#Verify data information and structure    
data_2020_cleaned.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1052 entries, 0 to 1054
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company_Brand  1052 non-null   object 
 1   Founded        1052 non-null   float64
 2   HeadQuarter    1052 non-null   object 
 3   Sector         1052 non-null   object 
 4   What_it_does   1052 non-null   object 
 5   Founders       1052 non-null   object 
 6   Investor       1052 non-null   object 
 7   Amount         1052 non-null   float64
 8   Stage          1052 non-null   object 
 9   column10       2 non-null      object 
dtypes: float64(2), object(8)
memory usage: 90.4+ KB


In [14]:
# Call DataFrame to understand DataFrame details for 2021.
query= "SELECT * FROM dbo.LP1_startup_funding2021"
data_2021 =pd.read_sql(query, connection)

data_2021.head()

Company_Brand  Founded HeadQuarter          Sector  \
0  Unbox Robotics   2019.0   Bangalore      AI startup   
1          upGrad   2015.0      Mumbai          EdTech   
2     Lead School   2012.0      Mumbai          EdTech   
3         Bizongo   2015.0      Mumbai  B2B E-commerce   
4        FypMoney   2021.0    Gurugram         FinTech   

                                        What_it_does  \
0  Unbox Robotics builds on-demand AI-driven ware...   
1     UpGrad is an online higher education platform.   
2  LEAD School offers technology based school tra...   
3  Bizongo is a business-to-business online marke...   
4  FypMoney is Digital NEO Bank for Teenagers, em...   

                                            Founders  \
0                        Pramod Ghadge, Shahid Memon   
1  Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...   
2                         Smita Deorah, Sumeet Mehta   
3            Aniket Deb, Ankit Tomar, Sachin Agrawal   
4                                      Kapil Banwari   

                                        Investor        Amount         Stage  
0                    BEENEXT, Entrepreneur First    $1,200,000  Pre-series A  
1       Unilazer Ventures, IIFL Asset Management  $120,000,000          None  
2               GSV Ventures, Westbridge Capital   $30,000,000      Series D  
3                         CDC Group, IDG Capital   $51,000,000      Series C  
4  Liberatha Kallat, Mukesh Yadav, Dinesh Nagpal    $2,000,000          Seed

In [15]:
#check the shape
data_2021.shape    

(1209, 9)

In [16]:
# check data information and structure
data_2021.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1209 entries, 0 to 1208
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company_Brand  1209 non-null   object 
 1   Founded        1208 non-null   float64
 2   HeadQuarter    1208 non-null   object 
 3   Sector         1209 non-null   object 
 4   What_it_does   1209 non-null   object 
 5   Founders       1205 non-null   object 
 6   Investor       1147 non-null   object 
 7   Amount         1206 non-null   object 
 8   Stage          781 non-null    object 
dtypes: float64(1), object(8)
memory usage: 85.1+ KB


#### Observation
The 'Amount' column, which represents financial figures, is currently of type object, indicating it contains string data.

### Investigate Why Amount Dtype is object instead of float

In [17]:
# Display the 'Amount' column
'''The currency symbol is a contributing factor to wrong Dtype instead of float'''

print(data_2021['Amount'])

0         $1,200,000
1       $120,000,000
2        $30,000,000
3        $51,000,000
4         $2,000,000
            ...     
1204        $3000000
1205       $20000000
1206       $55000000
1207       $26000000
1208        $8000000
Name: Amount, Length: 1209, dtype: object


In [18]:
# Removing dollar signs and commas
data_2021['Amount'] = data_2021['Amount'].str.replace('[\$,]', '', regex=True)

# Display the 'Amount' column
print(data_2021['Amount'])

0         1200000
1       120000000
2        30000000
3        51000000
4         2000000
          ...    
1204      3000000
1205     20000000
1206     55000000
1207     26000000
1208      8000000
Name: Amount, Length: 1209, dtype: object


In [19]:
# Confirm if all values under Amount are all numeric
data_2021['Amount'].apply(lambda x: isinstance(x, str) and not x.replace(',', '').replace('$', '').isdigit())


0       False
1       False
2       False
3       False
4       False
        ...  
1204    False
1205    False
1206    False
1207    False
1208    False
Name: Amount, Length: 1209, dtype: bool

In [20]:
# Filter Non-Numeric Values:
non_numeric_rows=data_2021[data_2021['Amount'].apply(lambda x: isinstance(x, str) and not x.replace(',', '').replace('$', '').isdigit())]

print(non_numeric_rows)


              Company_Brand  Founded HeadQuarter                    Sector  \
7               Qube Health   2016.0      Mumbai                HealthTech   
8                  Vitra.ai   2020.0   Bangalore              Tech Startup   
21                    Uable   2020.0   Bangalore                    EdTech   
39                 TruNativ   2019.0      Mumbai          Food & Beverages   
54                   AntWak   2019.0   Bangalore                    EdTech   
...                     ...      ...         ...                       ...   
1148              Godamwale   2016.0      Mumbai  Logistics & Supply Chain   
1160  Atomberg Technologies   2012.0      Mumbai      Consumer Electronics   
1161        Genext Students   2013.0      Mumbai                    EdTech   
1166              OckyPocky   2015.0    Gurugram                    EdTech   
1193        Sapio Analytics   2019.0      Mumbai         Computer Software   

                                           What_it_does  \
7   

#### Amount Observation
The column Amount still contains non-numeric values with undesclosed amount in some rows, hence the reason for object Dtype instead of float

In [21]:
# Replace "Undisclosed" with NaN in the 'Amount' column
data_2021['Amount'] = data_2021['Amount'].replace('Undisclosed', np.nan)

# Filter non-numeric values in the 'Amount' column
non_numeric_values = data_2021[~data_2021['Amount'].apply(lambda x: isinstance(x, (int, float)))]['Amount']


'''Print to confirm after replacing Undisclosed amount with NaN the right Dtype
Resurlt still indicate the wrong dtype and needs to be investigated further'''

# Print non-numeric values
print("Non-numeric values after replacing 'Undisclosed':")
print(non_numeric_values)

Non-numeric values after replacing 'Undisclosed':
0         1200000
1       120000000
2        30000000
3        51000000
4         2000000
          ...    
1204      3000000
1205     20000000
1206     55000000
1207     26000000
1208      8000000
Name: Amount, Length: 1093, dtype: object


#### Further investigation on the 'Amount' column

In [22]:
# Print unique non-numeric values, if any
if not non_numeric_values.empty:
    print("Non-numeric values after replacing 'Undisclosed':")
    print(non_numeric_values.unique())
else:
    print("No non-numeric values found after replacing 'Undisclosed'.")

Non-numeric values after replacing 'Undisclosed':
['1200000' '120000000' '30000000' '51000000' '2000000' '188000000'
 '200000' '1000000' '3000000' '100000' '700000' '9000000' '40000000'
 '49000000' '400000' '300000' '25000000' '160000000' '150000' '1800000'
 '5000000' '850000' '53000000' '500000' '1100000' '6000000' '800000'
 '10000000' '21000000' '7500000' '26000000' '7400000' '1500000' '600000'
 '800000000' '17000000' '3500000' '15000000' '215000000' '2500000'
 '350000000' '5500000' '83000000' '110000000' '500000000' '65000000'
 '150000000000' '300000000' '2200000' '35000000' '140000000' '4000000'
 '13000000' None '9500000' '8000000' 'Upsparks' '12000000' '1700000'
 '150000000' '100000000' '225000000' '6700000' '1300000' '20000000'
 '250000' '52000000' '3800000' '17500000' '42000000' '2300000' '7000000'
 '450000000' '28000000' '8500000' '37000000' '370000000' '16000000'
 '44000000' '770000' '125000000' '50000000' '4900000' '145000000'
 '22000000' '70000000' '6600000' '32000000' '2400

#### its observed some numeric values are encoded as strings with some few non-numeric values

In [23]:
# Define function to convert numeric strings to numeric values
def convert_to_numeric(value):
    try:
        return float(value)
    except (ValueError, TypeError):
        return np.nan

# Replace numeric strings with their corresponding numeric values
data_2021['Amount'] = data_2021['Amount'].apply(convert_to_numeric)

# Print the modified DataFrame
print(data_2021)

       Company_Brand  Founded HeadQuarter                 Sector  \
0     Unbox Robotics   2019.0   Bangalore             AI startup   
1             upGrad   2015.0      Mumbai                 EdTech   
2        Lead School   2012.0      Mumbai                 EdTech   
3            Bizongo   2015.0      Mumbai         B2B E-commerce   
4           FypMoney   2021.0    Gurugram                FinTech   
...              ...      ...         ...                    ...   
1204        Gigforce   2019.0    Gurugram  Staffing & Recruiting   
1205          Vahdam   2015.0   New Delhi       Food & Beverages   
1206    Leap Finance   2019.0   Bangalore     Financial Services   
1207    CollegeDekho   2015.0    Gurugram                 EdTech   
1208          WeRize   2019.0   Bangalore     Financial Services   

                                           What_it_does  \
0     Unbox Robotics builds on-demand AI-driven ware...   
1        UpGrad is an online higher education platform.   
2     

#### Verify the right dtype conversion on Amount 

In [24]:
# Check and print the data type of the 'Amount' column
print("Data type of 'Amount' column:", data_2021['Amount'].dtype)

Data type of 'Amount' column: float64


In [60]:
# Confirm data quality and consistency in Dtype conversion
data_2021.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1209 entries, 0 to 1208
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Company       1209 non-null   object 
 1   Founded       1208 non-null   float64
 2   HeadQuarter   1208 non-null   object 
 3   Sector        1209 non-null   object 
 4   What_it_does  1209 non-null   object 
 5   Founders      1205 non-null   object 
 6   Investor      1147 non-null   object 
 7   Amount        1056 non-null   float64
 8   Stage         781 non-null    object 
dtypes: float64(2), object(7)
memory usage: 85.1+ KB


#### Handle Missing values in data_2021

In [53]:
# Calculate missing values and their percentage
missing_values = data_2021.isnull().sum()
missing_percentage = (missing_values / len(data_2021)) * 100

# Print missing values with percentage
missing_info = pd.DataFrame({'Missing Values': missing_values, 'Percentage': missing_percentage})
print(missing_info)


              Missing Values  Percentage
Company                    0    0.000000
Founded                    1    0.082713
HeadQuarter                1    0.082713
Sector                     0    0.000000
What_it_does               0    0.000000
Founders                   4    0.330852
Investor                  62    5.128205
Amount                   153   12.655087
Stage                    428   35.401158


#### By observation, very little values are missing in all the columns, exception the column Stage with 35.40% missing values compared to the rest which has over 80% of its  values entered.

Rows with missing values less than 15% will be dropped to ensure data integrity and maintain a high-quality dataset. For columns with missing values exceeding 15%, such as the 'Stage' column with a substantial 35.40% missing values, will be populated rather than drop them outright. The decision is made to retain as much data as possible while still ensuring that the 'Stage' information, despite its high rate of missingness, is available for analysis.







In [56]:
from sklearn.impute import SimpleImputer

# Assuming data_2021 is your DataFrame

# Identify columns with missing values less than 15%
columns_to_drop = data_2021.columns[data_2021.isnull().sum() / len(data_2021) < 0.15]

# Drop rows with missing values in those columns
data_2021_dropped = data_2021.dropna(subset=columns_to_drop)

# Fill missing values in the 'Stage' column with mode
stage_imputer = SimpleImputer(strategy='most_frequent')
data_2021_dropped['Stage'] = stage_imputer.fit_transform(data_2021_dropped[['Stage']]).ravel()  # Flatten the 2D array

# Verify the changes
data_2021_dropped.info()


<class 'pandas.core.frame.DataFrame'>
Index: 994 entries, 0 to 1208
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Company       994 non-null    object 
 1   Founded       994 non-null    float64
 2   HeadQuarter   994 non-null    object 
 3   Sector        994 non-null    object 
 4   What_it_does  994 non-null    object 
 5   Founders      994 non-null    object 
 6   Investor      994 non-null    object 
 7   Amount        994 non-null    float64
 8   Stage         664 non-null    object 
dtypes: float64(2), object(7)
memory usage: 77.7+ KB


#### Check for duplicate

In [57]:
# Check the number of duplicate rows
num_duplicates = data_2021_dropped.duplicated().sum()

# Print the number of duplicate rows
print("Number of duplicate rows:", num_duplicates)

Number of duplicate rows: 15


In [58]:
# Drop duplicate rows
data_2021_no_duplicates = data_2021.drop_duplicates()

# Drop rows with missing values
data_2021_cleaned = data_2021_no_duplicates.dropna()

# Verify the changes
data_2021_cleaned.info()

<class 'pandas.core.frame.DataFrame'>
Index: 654 entries, 0 to 1208
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Company       654 non-null    object 
 1   Founded       654 non-null    float64
 2   HeadQuarter   654 non-null    object 
 3   Sector        654 non-null    object 
 4   What_it_does  654 non-null    object 
 5   Founders      654 non-null    object 
 6   Investor      654 non-null    object 
 7   Amount        654 non-null    float64
 8   Stage         654 non-null    object 
dtypes: float64(2), object(7)
memory usage: 51.1+ KB


#### Load csv data from other sources for analysis

In [25]:
# Read 2018 DataFrame to understand data structure.
data_2018=pd.read_csv("D:\\JHanson\\Justice Hanson\\DS Career Accelerator\Project 1\\Indian-Start-up-Investment-Analysis\\CSV Data\\startup_funding2018.csv")

data_2018.head(5)

Company Name                                           Industry  \
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

  Round/Series       Amount                          Location  \
0         Seed       250000       Bangalore, Karnataka, India   
1         Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2     Series A  ₹65,000,000           Gurgaon, Haryana, India   
3        Angel      2000000       Noida, Uttar Pradesh, India   
4         Seed            —  Hyderabad, Andhra Pradesh, India   

                                       About Company  
0  TheCollegeFever is a hub for fun, fiesta and f...  
1  A startup which aggregates milk from dairy far...  
2          Leading Online Loans Marketplace in India  
3  PayMe India is an innovative FinTech organizat...  
4  Eunimart is a one stop solution for merchants ...

In [26]:
# Read 2019 DataFrame to understand data structure.
data_2019=pd.read_csv("D:\\JHanson\\Justice Hanson\\DS Career Accelerator\Project 1\\Indian-Start-up-Investment-Analysis\\CSV Data\\startup_funding2019.csv")

data_2019.head(5)

Company/Brand  Founded HeadQuarter           Sector  \
0  Bombay Shaving      NaN         NaN        Ecommerce   
1       Ruangguru   2014.0      Mumbai           Edtech   
2        Eduisfun      NaN      Mumbai           Edtech   
3        HomeLane   2014.0     Chennai  Interior design   
4        Nu Genes   2004.0   Telangana         AgriTech   

                                        What it does  \
0         Provides a range of male grooming products   
1  A learning platform that provides topic-based ...   
2            It aims to make learning fun via games.   
3              Provides interior designing solutions   
4  It is a seed company engaged in production, pr...   

                                Founders  \
0                     Shantanu Deshpande   
1  Adamas Belva Syah Devara, Iman Usman.   
2                          Jatin Solanki   
3           Srikanth Iyer, Rama Harinath   
4                 Narayana Reddy Punyala   

                                            Investor     Amount($)  \
0                               Sixth Sense Ventures    $6,300,000   
1                                   General Atlantic  $150,000,000   
2     Deepak Parekh, Amitabh Bachchan, Piyush Pandey   $28,000,000   
3  Evolvence India Fund (EIF), Pidilite Group, FJ...   $30,000,000   
4           Innovation in Food and Agriculture (IFA)    $6,000,000   

           Stage  
0            NaN  
1       Series C  
2  Fresh funding  
3       Series D  
4            NaN

#### Print the original column names to verify their structure

In [27]:
# Inspect column names in each DataFrame  to understand data structure.

print("Column names in 2018 DataFrame:")
print(data_2018.columns)

print("\nColumn names in 2019 DataFrame:")
print(data_2019.columns)

print("\nColumn names in 2020 DataFrame:")
print(data_2020.columns)

print("\nColumn names in 2021 DataFrame:")
print(data_2021.columns)

Column names in 2018 DataFrame:
Index(['Company Name', 'Industry', 'Round/Series', 'Amount', 'Location',
       'About Company'],
      dtype='object')

Column names in 2019 DataFrame:
Index(['Company/Brand', 'Founded', 'HeadQuarter', 'Sector', 'What it does',
       'Founders', 'Investor', 'Amount($)', 'Stage'],
      dtype='object')

Column names in 2020 DataFrame:
Index(['Company_Brand', 'Founded', 'HeadQuarter', 'Sector', 'What_it_does',
       'Founders', 'Investor', 'Amount', 'Stage', 'column10'],
      dtype='object')

Column names in 2021 DataFrame:
Index(['Company_Brand', 'Founded', 'HeadQuarter', 'Sector', 'What_it_does',
       'Founders', 'Investor', 'Amount', 'Stage'],
      dtype='object')


#### Column name Observations:
The datasets from 2018 to 2021 on Indian startup funding contain varying column names. The inconsistent column names necessitates a renaming strategy to align them uniformly before effectively merged into one DataFrame

### Column Mapping

In [28]:
# Define a column_mapping dictionary to standardize column names for consistent and accurate analysis
column_mapping = {
    'Company Name': 'Company',
    'Company/Brand': 'Company',
    'Company_Brand': 'Company',
    'Industry': 'Sector',
    'Sector': 'Sector',
    'Round/Series': 'Stage',
    'Stage': 'Stage',
    'Amount': 'Amount',
    'Amount($)': 'Amount',
    'Location': 'HeadQuarter',
    'HeadQuarter': 'HeadQuarter',
    'About Company': 'What_it_does',
    'What it does': 'What_it_does',
    'What_it_does': 'What_it_does',
    'Founded': 'Founded',
    'Founders': 'Founders',
    'Investor': 'Investor'
}

# Rename the columns in each DataFrame using the rename method and the column_mapping
data_2018.rename(columns=column_mapping, inplace=True)
data_2019.rename(columns=column_mapping, inplace=True)
data_2020.rename(columns=column_mapping, inplace=True)
data_2021.rename(columns=column_mapping, inplace=True)

# Print the renamed column names for verification
print("\nRenamed column names:")
print("2018 DataFrame:", data_2018.columns)
print("2019 DataFrame:", data_2019.columns)
print("2020 DataFrame:", data_2020.columns)
print("2021 DataFrame:", data_2021.columns)


Renamed column names:
2018 DataFrame: Index(['Company', 'Sector', 'Stage', 'Amount', 'HeadQuarter', 'What_it_does'], dtype='object')
2019 DataFrame: Index(['Company', 'Founded', 'HeadQuarter', 'Sector', 'What_it_does',
       'Founders', 'Investor', 'Amount', 'Stage'],
      dtype='object')
2020 DataFrame: Index(['Company', 'Founded', 'HeadQuarter', 'Sector', 'What_it_does',
       'Founders', 'Investor', 'Amount', 'Stage', 'column10'],
      dtype='object')
2021 DataFrame: Index(['Company', 'Founded', 'HeadQuarter', 'Sector', 'What_it_does',
       'Founders', 'Investor', 'Amount', 'Stage'],
      dtype='object')


### Merge 2018, 2019, 2021, 2021 DataFrame into one table (df_Merged)

In [29]:
# Merge DataFrames into one table using the concatenation function.
df_merged = pd.concat([data_2018, data_2019, data_2020, data_2021], ignore_index=True)

# Print the first few rows of the merged DataFrame
print("\nMerged DataFrame:")
print(df_merged.head())


Merged DataFrame:
           Company                                             Sector  \
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

      Stage       Amount                       HeadQuarter  \
0      Seed       250000       Bangalore, Karnataka, India   
1      Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2  Series A  ₹65,000,000           Gurgaon, Haryana, India   
3     Angel      2000000       Noida, Uttar Pradesh, India   
4      Seed            —  Hyderabad, Andhra Pradesh, India   

                                        What_it_does  Founded Founders  \
0  TheCollegeFever is a hub for fun, fiesta and f...      NaN      NaN   
1  A startup which ag

### Introduce new column (year_founded) for easy comparison and time series analysis

In [30]:
# Add Year_founded column to each DataFrame for comparison.
data_2018['year_founded'] = 2018
data_2019['year_founded'] = 2019
data_2020['year_founded'] = 2020
data_2021['year_founded'] = 2021

In [31]:
# Merge all DataFrames into one.
df_merged = pd.concat([data_2018, data_2019, data_2020, data_2021], ignore_index=True)

### Verify if new column (year_founded) is added to merged DataFrame successfully

In [32]:
# Print the first few rows of the merged DataFrame to verify that the year_founded column has been added correctly.

df_merged.head(5)

Company                                             Sector  \
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

      Stage       Amount                       HeadQuarter  \
0      Seed       250000       Bangalore, Karnataka, India   
1      Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2  Series A  ₹65,000,000           Gurgaon, Haryana, India   
3     Angel      2000000       Noida, Uttar Pradesh, India   
4      Seed            —  Hyderabad, Andhra Pradesh, India   

                                        What_it_does  year_founded  Founded  \
0  TheCollegeFever is a hub for fun, fiesta and f...          2018      NaN   
1  A startup which aggregates milk from dairy far...          2018      NaN   
2          Leading Online Loans Marketplace in India          2018      NaN   
3  PayMe India is an innovative FinTech organizat...          2018      NaN   
4  Eunimart is a one stop solution for merchants ...          2018      NaN   

  Founders Investor column10  
0      NaN      NaN      NaN  
1      NaN      NaN      NaN  
2      NaN      NaN      NaN  
3      NaN      NaN      NaN  
4      NaN      NaN      NaN

### Verify from the bottom of the merged DataFrame

In [33]:
# Print the last few rows of the merged DataFrame to verify years have been added correctly

df_merged.tail(5)

Company                 Sector         Stage      Amount  \
2874      Gigforce  Staffing & Recruiting  Pre-series A   3000000.0   
2875        Vahdam       Food & Beverages      Series D  20000000.0   
2876  Leap Finance     Financial Services      Series C  55000000.0   
2877  CollegeDekho                 EdTech      Series B  26000000.0   
2878        WeRize     Financial Services      Series A   8000000.0   

     HeadQuarter                                       What_it_does  \
2874    Gurugram                  A gig/on-demand staffing company.   
2875   New Delhi  VAHDAM is among the world’s first vertically i...   
2876   Bangalore  International education loans for high potenti...   
2877    Gurugram  Collegedekho.com is Student’s Partner, Friend ...   
2878   Bangalore  India’s first socially distributed full stack ...   

      year_founded  Founded                       Founders  \
2874          2021   2019.0    Chirag Mittal, Anirudh Syal   
2875          2021   2015.0                     Bala Sarda   
2876          2021   2019.0     Arnav Kumar, Vaibhav Singh   
2877          2021   2015.0                   Ruchir Arora   
2878          2021   2019.0  Vishal Chopra, Himanshu Gupta   

                              Investor column10  
2874                   Endiya Partners      NaN  
2875                          IIFL AMC      NaN  
2876                      Owl Ventures      NaN  
2877  Winter Capital, ETS, Man Capital      NaN  
2878    3one4 Capital, Kalaari Capital      NaN

 ### Data Understanding and Preparations

Preprocess df_merged to check data quality

In [34]:
# check the shape of the merged DataFrame to understand number of rows and columns

df_merged.shape


(2879, 11)

In [35]:
# Get the information about the merged DataFrame

df_merged.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2879 entries, 0 to 2878
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Company       2879 non-null   object 
 1   Sector        2874 non-null   object 
 2   Stage         2405 non-null   object 
 3   Amount        2726 non-null   object 
 4   HeadQuarter   2859 non-null   object 
 5   What_it_does  2879 non-null   object 
 6   year_founded  2879 non-null   int64  
 7   Founded       2323 non-null   float64
 8   Founders      2346 non-null   object 
 9   Investor      2291 non-null   object 
 10  column10      2 non-null      object 
dtypes: float64(1), int64(1), object(9)
memory usage: 247.5+ KB


In [36]:
# Verify why Amount is object

df_merged['Amount']

0            250000
1       ₹40,000,000
2       ₹65,000,000
3           2000000
4                 —
           ...     
2874      3000000.0
2875     20000000.0
2876     55000000.0
2877     26000000.0
2878      8000000.0
Name: Amount, Length: 2879, dtype: object

#### Observation

The Amount column is of type object. This is because the values in this column are in the form of strings. This means that the values in this column are not numeric and need to be converted to numeric data types.

In [37]:
# Set to store unique errors that may be encountered during conversion process.
unique_errors = set()

# Function to clean and convert amounts
def convert_amount(amount):
    try:
        # Remove leading and trailing currency symbols ($ and ₹) and commas
        cleaned_amount = amount.replace('$', '').replace('₹', '').replace(',', '')
        #convert the cleaned amount to a float and returns it.
        return float(cleaned_amount)
    except ValueError:
        # Store unique errors
        unique_errors.add(amount)
        return np.nan

# Apply conversion function to the 'Amount' column
df_merged['Amount'] = df_merged['Amount'].apply(lambda x: convert_amount(str(x)) if pd.notnull(x) else x)

# Print unique errors
for error in unique_errors:
    print(f"Unique Error: {error}")


Unique Error: Undisclosed
Unique Error: —


#### Verify Data type for Amount column

In [38]:
# Check data information again  
df_merged.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2879 entries, 0 to 2878
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Company       2879 non-null   object 
 1   Sector        2874 non-null   object 
 2   Stage         2405 non-null   object 
 3   Amount        2566 non-null   float64
 4   HeadQuarter   2859 non-null   object 
 5   What_it_does  2879 non-null   object 
 6   year_founded  2879 non-null   int64  
 7   Founded       2323 non-null   float64
 8   Founders      2346 non-null   object 
 9   Investor      2291 non-null   object 
 10  column10      2 non-null      object 
dtypes: float64(2), int64(1), object(8)
memory usage: 247.5+ KB


#### Confirm Amount column is rightly formatted

In [39]:
# Verify why Amount is object

df_merged['Amount']

0         250000.0
1       40000000.0
2       65000000.0
3        2000000.0
4              NaN
           ...    
2874     3000000.0
2875    20000000.0
2876    55000000.0
2877    26000000.0
2878     8000000.0
Name: Amount, Length: 2879, dtype: float64